homeassistant/components/telegram_bot/bot.py

```diff
@@ -762,9 +762,17 @@ async def send_file(
                         message_thread_id=params[ATTR_MESSAGE_THREAD_ID],
                         context=context,
                     )
-
-                msg_ids[chat_id] = msg.id
-                file_content.seek(0)
+                if msg is not None:
+                    msg_ids[chat_id] = msg.id
+                    file_content.seek(0)
+                else:
+                    _LOGGER.error(
+                        "Failed to send %s to chat_id %s: message object is None. "
+                        "If using Markdown v2 parse mode, ensure all special characters "
+                        "are properly escaped",
+                        file_type,
+                        chat_id,
+                    )
```

Comment:
The moment we return `None` we already logged what happened
https://github.com/home-assistant/core/pull/153911/files#diff-136482d5ac1aa6dd8cae166601b3840a6012b9875580143482218ce3b5ba1f13L466-L470